Hijack pr 363 (#416)
NINAnor/ecRxiv: R/NO_ALIE_002.qmd

# and in the same order that corresponds to the indicator IDs,  
# to the meta data fields where the indicators differ. 
# For example, if the indicators cover different ecosystem types, 
# add the ecosystems as a list under 'Ecosystems', but if they are in the
# same realm, same country etc., then you only need to write the realm and
# country once. 
indicatorName: 
  - Alien plants
verbatimName: 
  - Fremmede plantearter # the indicator name that us used in the native language
folderName: NO_ALIE_002 # Leave to be filled out by ecRxiv during typesetting (name of the folder under 'indicators/'). 
# If it's a single indicator being reported, the folder name should be the same as the indicatorID. 
# If there are multiple ID's then the folder name could look something like XX_YYY_ZZZ-ÅÅÅ.
country: Norway
continent: Europe
ECT: B1 - Biotic compositional characteristics
Realm: # Please copy and paste Realms, Biomes and Ecostsem names from here: https://github.com/NINAnor/ecRxiv/wiki#ecosystem-typology 
  - Terrestrial (T)
Biome: 
  - T6 Polar-alpine (cryogenic) biome
Ecosystem: 
  - T6.5 Trop alpine grassland
  - T6.3 Polar tundra
verbatimEcosystem:
  - Fjell
yearAdded: 2025

> yearLastUpdate: 2026

Reviewer: Update and add _common.R. The citation in the html shows 2025 as the year, but in the new _common.R this is fixed so that it shows yearLastUpdate

> status: incomplete # options are complete, incomplete, or deprecated

Reviewer: switch to complete

Normalised: Yes # Yes or no depending on whether the metric has been normliased or not.
SpatialAggregationPathway: Scale/Truncate - Aggregate (site-level arithmetic mean; cluster bootstrap) # e.g. Scale - Transform - Aggregate (length-weighted arithmetic mean) - Truncate
Precision: 95% confidence intervals estimated using a stratified cluster bootstrap of ANO sites. # What kind of quantitative unceratinties are reported alongside the point estimates. Can be either inferrential (i.e. bootstrapped confidence intervals or standard errors) or descriptive (variation, standard deviation or percentiles from the distribution of the raw indicator values).
format: 
  html:
    embed-resources: true
    code-fold: true
    toc: true
    toc-title: Contents
    toc-depth: 3
    smooth-scroll: true
execute: 
  cache: true
author:
  - name: Joachim Töpper             # Enter name
    email: joachim.topper@nina.no  # Enter email
    affiliations:
      - id: myID
        name: Norwegian Institute for Nature Research # Enter affiliations
AuthorList: Töpper, J.
date: July 1, 2026 # Enter date
data_availability: silver

> code_reproducibility: silver

Reviewer: with renv we can go to gold

```

```{r licenseBadge}
#| echo: false
# The following inserts a badge specifying the GPL GNU version 3 license. ecRxiv recommends using this license, but you can decide to use another. Contact ecRxiv for help in insert a badge for a different license
license_badge()
```

::: 
 

> **Recommended citation**: `r paste(auth, " ", year, ". ", name, " (ID(s): ", folder_name, ") ", "v. ", version, ". ecRxiv: ", url, sep="")`

::: {.callout-tip collapse="true"}
## Metadata
```{r tbl-meta2}
#| tbl-cap: 'Indicator metadata'
#| echo: false
#| warning: false
meta |>
  dplyr::filter(Variable %in% c(
    "Indicator ID",
    "Indicator Name",
    "verbatimName",
    "Continent",
    "Country",
    "Ecosystem Condition Typology Class", 
    "Realm",
    "Biome",
    "Ecosystem",
    "verbatimEcosystem",
    "Year added", 
    "Last update",
    "Version",
    "Version comment",
    "Normalised",
    "Spatial aggregation pathway",
    "Precision")) |>   
  kable()

```
:::

::: {.callout-tip collapse="true"}

## Log

<!--# Update this logg with short messages for each update -->
- 21 May 2026 - original start-up of the 2026 alien species indicator for mountains

> - 31^st July 2026 - Accepted

Reviewer: the ^ don't render correctly. I've just been using 31 June 2026. And it June, not July

:::

<hr />

<!--# Document you work below.  -->

## 1. Summary

> 1. **Background and rationale**: The 'Alien plant species' indicator describes how abundant alien plant species are in an ecosystem on a scale from 0-100% cover. Alien plant species have a profound impact on native species and ecosystem dynamics and thus affect ecosystem condition significantly. The main anthropogenic factors behind increasing alien plant species abundance are i) introduction of alien plant species, and ii) climate change. See Framstad et al. (2022) for previous implementations of this indicator.

Reviewer: ```suggestion
1. **Background and rationale**: Alien plant species have a profound impact on native species and ecosystem dynamics and thus affect ecosystem condition significantly. This indicator describes how abundant alien plant species are in an ecosystem. The main anthropogenic factors behind increasing alien plant species abundance are i) introduction of alien plant species, and ii) climate change. See Framstad et al. (2022) for previous implementations of this indicator.
```

> 2. **Methods**: The indicator uses ANO monitoring data on alien plant species cover. The reference values for scaling the original cover values to a condition index are as follows:

Reviewer: ```suggestion
2. **Methods**: The indicator uses ANO monitoring data on alien plant species cover, originally recorded on a scale from 0-100% cover. The definition of an alien plant species are species introduced by humans and which we not established in Norway before year 1800. The reference values for scaling the original cover values to a condition index are as follows:
```

> - x100 (best possible condition) is set to 0% alien plant species cover and scales to an indicator value of 1

Reviewer: ```suggestion
 ~x~100 (best possible condition) is set to 0% alien plant species cover and scales to an indicator value of 1, and; 
```

> - x0 (lowest possible condition) is set to 50% and higher alien species cover and scales to an indicator value of 0

Reviewer: ```suggestion
 ~x~0 (lowest possible condition) is set to 50% and higher alien species cover and scales to an indicator value of 0.
```

### 2.2 Sampling design
At every ANO site, the indicator variable is registered at up to 18 monitoring points, dependent on the ecosystem designation.

### 2.3 Original units
Abundance data for alien plant species in ANO is given as % cover within a 250 m^2^ sample circle following the national alien species list (Artsdatabanken 2023).

### 2.4 Instructions for citing, using and accessing data
The ANO data are openly available from [Miljodirektoratet](https://kartkatalog.miljodirektoratet.no/Dataset/Details/2054).

### 2.5 Additional comments about the dataset
The ANO program is run by the Norwegian Environment Agency. On their [web site](https://www.miljodirektoratet.no/ansvarsomrader/overvaking-arealplanlegging/miljoovervaking/overvakingsprogrammer/natur-pa-land/arealrepresentativ-naturovervakning-ano/) you may also find some additional summary data. 

## 3. Indicator properties

### 3.1 Ecosystem Condition Typology Class (ECT)
The 'Alien plant species' indicator reflects the plant species composition and thus represents ECT class 'B1 Biotic compositional characteristics'.

### 3.2 Ecosystem condition characteristic

> The 'Alien plant species' indicator addresses both the ecosystem characteristic 'functionally important species and biophysical structures' and 'biological diversity' in the Norwegian framework for ecological condition. The indicator has been linked solely to the former in the previous assessments of ecological condition in forest and mountain ecosystems.

Reviewer: There's been some misunderstanding about what this header is ment to be
```suggestion
Intact native vegetation
```

#summary(as.factor(ANO.mount$ano_punkt_id))
# there's many double presences, probably some wrong registrations of point numbers,

## add region information
nor <- st_read(here::here("data/outlineOfNorway_EPSG25833.shp"),
               quiet = T)%>%
  st_as_sf() %>%
  st_transform(crs = st_crs(ANO.mount))

reg <- st_read(here::here("data/regions.shp"),
               quiet = T) %>%
  st_as_sf() %>%
  st_transform(crs = st_crs(ANO.mount))

# change region names to something R-friendly
# reg$region
reg$region <- c("Northern.Norway","Central.Norway","Eastern.Norway","Western.Norway","Southern.Norway")

regnor <- st_intersection(reg,nor)

ANO.mount = st_join(ANO.mount, regnor, left = TRUE, join = st_nearest_feature)

```

We can illustrate the cover of alien plant species in ANO points in Norway's five regions in a figure
```{r, warning=F}
ggplot(ANO.mount, aes(x = region, y = fremmedarter_total_dekning)) +
  geom_boxplot() +
  labs(
    x = "Region",
    y = "Alien plant species cover (%)"
    )
```

#### Normalisation
We use a normalisation function scaling linearly between the reference values. The function is demonstrated in the code below, and in @fig-demo.

```{r fig-demo}

> #| fig-cap: 'Normalisation function (piecewise linear)'

Reviewer: ```suggestion
#| fig-cap: 'Normalisation function'
```

Technically it's more a linear scaling (no transformation) with truncation

## 10. Results

Indicator values are calculated as arithmetic means of the normalised indicator values. To quantify uncertainty, we perform a stratified cluster bootstrap with 1,000 replicates. Within each replicate, ANO sites are resampled with replacement within each region while retaining all monitoring points from each selected site. The distribution of bootstrap estimates is used to calculate the mean indicator value and the associated 95% confidence interval.

```{r}

> set.seed(123)

Reviewer: This cluster sampling is good I think. Treating the locations as the sampling unsits, and asking, what if we had sampled different areas. The distribution becomes an inferential probability distribution for the true indicator value at regional level.

B <- 1000

dat <- ANO.mount |>
  st_drop_geometry() |>
  filter(!is.na(alien_res), !is.na(region))

boot_results <- map_dfr(1:B, function(i) {
  
  boot_dat <- dat |>
    group_by(region) |>
    group_modify(~ {
      
      sites <- unique(.x$ano_flate_id)
      
      sampled_sites <- sample(
        sites,
        size = length(sites),
        replace = TRUE
      )
      
      bind_rows(
        lapply(sampled_sites, function(s) {
          filter(.x, ano_flate_id == s)
        })
      )
    }) |>

>     ungroup()
>   
>   regional <- boot_dat |>
>     group_by(region) |>
>     summarise(
>       sim = mean(alien_res, na.rm = TRUE),
>       .groups = "drop"
>     ) |>
>     rename(Region = region)
>   
>   national <- tibble(
>     Region = "Norway",
>     sim = mean(boot_dat$alien_res, na.rm = TRUE)
>   )

Reviewer: Should this be done in two steps to account for unbalanced number of points per location?

```suggestion
  regional <- boot_dat |>
    group_by(ano_flate_id) |>
    summarise(alien_XXX = mean(alien_res), .group = "drop") |>
    group_by(region) |>
    summarise(
      sim = mean(alien_XXX, na.rm = TRUE),
      .groups = "drop"
    ) |>
    rename(Region = region)
  
  national <- tibble(
    Region = "Norway",
    sim = mean(boot_dat$alien_res, na.rm = TRUE)
  )
```

```

## 11. Export file

<!--# 

Optional: Display the code (don't execute it) or the workflow for exporting the indicator values to file. Ideally the indicator values are exported as a georeferenced shape or raster file with indicators values, reference values and errors. You can also chose to export the raw (un-normalised or unscaled variable) as a separate product. You should not save large spatial output data on GitHub. You can use eval=FALSE to avoid code from being executed (example below - delete if not relevant) 

-->

```{r export}
#| eval: false

# write to parquet using arrow 
NO_ALIE_002_sim <- boot_results |>
  mutate(year = 2024)

arrow::write_parquet(
  NO_ALIE_002_sim,
  "data/NO_ALIE_002_sim.parquet"
)

```

## 12. Author contributions

<!-- Fill in CReDIT and MeRIT statments to clarify the contributions from individual authors. For MeRIT, see https://www.nature.com/articles/s41467-023-37039-1. For CRedit, see https://credit.niso.org/. You can start with the template structur below, and just delete the rows you dont use.  -->

**CRediT statement**

* Conceptualization: [JT]
* Methodology: [JT]
* Data curation: [JT]
* Formal analysis: [JT]
* Visualization: [JT]

> * Reproducibility: [JT]

Reviewer: ```suggestion
* Conceptualization: [JT]
* Methodology: [JT, MG]
* Data curation: [JT, MG]
* Formal analysis: [JT, MG]
* Visualization: [JT]
* Reproducibility: [JT, MG]
```

::: {.callout-tip collapse="true"}
## Session Info
<!-- You can leave this last part as it is, unless you have a good reason to alter it. -->

This information is mainly aimed at developers that need to recreate the R environment that produced the html.
Important note: this information is about the R session that was running when the html was rendered, and if some of the code chunks were not evaluated when the html was rendered, they may have been executed in a different environment.  

```{r sessionInfo}
#| echo: false

> sessionInfo()
> ```
> 

Reviewer: ```suggestion
## Session Info
This workflow uses `renv`. Run `renv::restore()` to get started.

```

# For example, if the indicators cover different ecosystem types, 
# add the ecosystems as a list under 'Ecosystems', but if they are in the
# same realm, same country etc., then you only need to write the realm and
# country once. 
indicatorName: 
  - Alien plants
verbatimName: 
  - Fremmede plantearter # the indicator name that us used in the native language
folderName: NO_ALIE_002 # Leave to be filled out by ecRxiv during typesetting (name of the folder under 'indicators/'). 
# If it's a single indicator being reported, the folder name should be the same as the indicatorID. 
# If there are multiple ID's then the folder name could look something like XX_YYY_ZZZ-ÅÅÅ.
country: Norway
continent: Europe
ECT: B1 - Biotic compositional characteristics
Realm: # Please copy and paste Realms, Biomes and Ecostsem names from here: https://github.com/NINAnor/ecRxiv/wiki#ecosystem-typology 
  - Terrestrial (T)
Biome: 
  - T6 Polar-alpine (cryogenic) biome
Ecosystem: 
  - T6.5 Trop alpine grassland
  - T6.3 Polar tundra
verbatimEcosystem:
  - Fjell
yearAdded: 2025
yearLastUpdate: 2026
status: incomplete # options are complete, incomplete, or deprecated

> Version: "2.0.0" # See wiki page: https://github.com/NINAnor/ecRxiv/wiki#versioning

Reviewer: ```suggestion
Version: "1.0.0" # See wiki page: https://github.com/NINAnor/ecRxiv/wiki#versioning
```

Reviewer: strange to go from 000.002 to 2.0.0

> VersionComment: separate code for mountain ecosystem. Revised reference values.

Reviewer: ```suggestion
VersionComment: separate code for mountain ecosystem. Revised reference values. Previous version was 000.002
```

conflicts_prefer(dplyr::lag)
conflicts_prefer(dplyr::select)
pull<-dplyr::pull

knitr::opts_chunk$set(echo = TRUE)
```

```{r source}
#| echo: false
#| cache: false
source(here::here("_common.R"))
```

<!--# The following parts are autogenerated. Do not edit. -->

::: {layout-ncol="6"}

```{r statusBadge}
#| echo: false
status_badge(as.character(st))
```

```{r versionBadge}
#| echo: false
version_badge(my_version_number = as.character(version[[1]]))
```

```{r dataBadge}
#| echo: false
data_badge(as.character(badges[[1]]))
```

```{r codeBadge}
#| echo: false
code_badge(as.character(badges[[2]]))
```

```{r openScienceBadge}
#| echo: false
open_science_badge(as.character(badges[[3]]))
```

```{r licenseBadge}
#| echo: false
# The following inserts a badge specifying the GPL GNU version 3 license. ecRxiv recommends using this license, but you can decide to use another. Contact ecRxiv for help in insert a badge for a different license
license_badge()
```

::: 
 

> **Recommended citation**: `r paste(auth, " ", year, ". ", name, " (ID(s): ", folder_name, ") ", "v. ", version, ". ecRxiv: ", url, sep="")`

::: {.callout-tip collapse="true"}
## Metadata
```{r tbl-meta2}

> #| tbl-cap: 'Indicator metadata'

Reviewer: needs updating

#| echo: false
#| warning: false
meta |>
  dplyr::filter(Variable %in% c(
    "Indicator ID",
    "Indicator Name",
    "Continent",
    "Country",
    "Ecosystem Condition Typology Class", 
    "Realm",
    "Biome",
    "Ecosystem",
    "Year added", 
    "Last update",
    "Version",
    "Version comment",
    "Normalised",
    "Spatial aggregation pathway")) |>   
  kable()

```
:::

::: {.callout-tip collapse="true"}

## Log

<!--# Update this logg with short messages for each update -->

> - 21 May 2026 - original start-up of the 2026 alien species indicator for mountains

Reviewer: add acceptance data

:::

> 

Reviewer: Add 'reviewed by'

alien_re_sim

# combining alien_no_sim and alien_re_sim
NO_ALIE_002_sim <- rbind(data.frame(region = 'Norway', sim = alien.no_sim), alien_re_sim)

```

## 11. Export file

<!--# 

Optional: Display the code (don't execute it) or the workflow for exporting the indicator values to file. Ideally the indicator values are exported as a georeferenced shape or raster file with indicators values, reference values and errors. You can also chose to export the raw (un-normalised or unscaled variable) as a separate product. You should not save large sptaial output data on GitHub. You can use eval=FALSE to avoid code from being executed (example below - delete if not relevant) 

-->

```{r export}
#| eval: false
```

## 12. Author contributions

<!-- Fill in CReDIT and MeRIT statments to clarify the contributions from individual authors. For MeRIT, see https://www.nature.com/articles/s41467-023-37039-1. For CRedit, see https://credit.niso.org/. You can start with the template structur below, and just delete the rows you dont use.  -->

**CRediT statement**

* Conceptualization: [JT, AK, EF, UJ]
* Methodology: [JT]
* Data curation: [JT]
* Formal analysis: [JT]
* Visualization: [JT]

> * Reproducibility: [AH]

Reviewer: I don't think it makes sense to add people to the CRediT if you're not authors.

Author: should I add those listed here to the author list? Or jsut keep JT?

Reviewer: Just keep JT I would argue.

               quiet = T) %>%
  st_as_sf() %>%
  st_transform(crs = st_crs(ANO.mount))

# change region names to something R-friendly
# reg$region
reg$region <- c("Northern.Norway","Central.Norway","Eastern.Norway","Western.Norway","Southern.Norway")

regnor <- st_intersection(reg,nor)

ANO.mount = st_join(ANO.mount, regnor, left = TRUE, join = st_nearest_feature)

```

We can illustrate the cover of alien plant species in ANO points in Norway's five regions in a figure
```{r, warning=F}
ggplot(ANO.mount, aes(x = region, y = fremmedarter_total_dekning)) +
  geom_boxplot() +
  labs(
    x = "Region",
    y = "Alien plant species cover (%)"
    )
```

#### Normalisation
We use a normalisation function scaling linearly between the reference values. The function is demonstrated in the code below, and in @fig-demo.

```{r fig-demo}
#| fig-cap: 'Normalisation function (piecewise linear)'

temp <- tibble(
  variable = 0:100
)

temp <- temp %>% mutate(indicator = ec_normalise(variable = variable,
                          x0 = 50,
                          #x60 = 5,
                          x100 = 0,
                          fun = "linear")
)

> temp |>

Reviewer: would be good to use ec_norm_plot() to make the figures similar to the others.

```{r, warning=F}
## Fit ordered beta regression null-model with a random effect
# Norway
alien.no <- glmmTMB(
  alien_res ~ 1 + (1|ano_flate_id),
  data = ANO.mount,
  family = ordbeta
)

summary(alien.no)

# regions
alien.re <- glmmTMB(
  alien_res ~ 0 + region + (1|ano_flate_id),
  data = ANO.mount,
  family = ordbeta
)

summary(alien.re)
```

## 10. Results

The model estimates and standard errors are on a logit scale, but we can calculate indicator values and confidence intervals (CI) by transforming back from logit to values between 0-1
```{r}

NO_ALIE_002_table <- tibble(
  Region = c("Norway", "Central Norway", "Eastern Norway", "Northern Norway", "Southern Norway", "Western Norway"),
  indicator_value = c(plogis(summary(alien.no)$coef$cond[,1]), plogis(summary(alien.re)$coef$cond[,1])),
  CI_upper = c(plogis(summary(alien.no)$coef$cond[,1]+1.96*summary(alien.no)$coef$cond[,2]), plogis(summary(alien.re)$coef$cond[,1]+1.96*summary(alien.re)$coef$cond[,2])),
  CI_lower = as.numeric(plogis(summary(alien.no)$coef$cond[,1]-1.96*summary(alien.no)$coef$cond[,2]), plogis(summary(alien.re)$coef$cond[,1]-1.96*summary(alien.re)$coef$cond[,2]))
)

> NO_ALIE_002_table

Reviewer: The output of this show that the indicator values are sometimes outside the 95% CIs.

```

To express the results as a distribution, we need to simulate mean estimates based on the models' intercept and associated standard error as well as the dispersion parameter.
```{r}

## Norway
# Fixed-effect intercept and SE
b0 <- fixef(alien.no)$cond["(Intercept)"]
se_b0 <- sqrt(vcov(alien.no)$cond["(Intercept)", "(Intercept)"])
# Dispersion parameter (For glmmTMB this is usually on the response/positive scale via sigma() )
phi <- sigma(alien.no)

# draw estimates and backtransform from logit-scale
eta_draws <- rnorm(1000, mean = b0, sd = se_b0)
mu_draws  <- plogis(eta_draws)

# draw from beta distribution to consider the dispersion parameter 
alien.no_sim <- rbeta(
  n = 1000,
  shape1 = mu_draws * phi,
  shape2 = (1 - mu_draws) * phi

> )

Reviewer: I think this should be dropped entirely. With mu_draws we get the probability distribution for the trua value of the parameter estimate. If we then sample values and predict, we are predicting new observed alues, and the precision estimate is back to being descriptive rather than inferential. You can reuse my code from NO_FUMO_004 #418 (pushing it soon)

Reviewer: It also samples the wrong distribution I think. rbeta doesn't include zeros and ones.

## regions
b_re <- fixef(alien.re)$cond
vc_re <- vcov(alien.re)$cond
phi <- sigma(alien.re)

eta_draws <- MASS::mvrnorm(
  n = 1000,
  mu = b_re,
  Sigma = vc_re
)

mu_draws <- plogis(eta_draws)

alien_re_sim <- do.call(
  rbind,
  lapply(seq_along(b_re), function(i) {
    
    region <- names(b_re)[i]
    mu_i <- mu_draws[, i]
    
    sim <- rbeta(
      n = 1000,
      shape1 = mu_i * phi,
      shape2 = (1 - mu_i) * phi
    )
    
    data.frame(
      region = region,
#      draw = 1:10,
#      eta = eta_draws[, i],
#      mu = mu_i,
#      dispersion = phi,
      sim = sim
    )
  })
)

alien_re_sim$region <- sub("^region", "", alien_re_sim$region)

head(alien_re_sim)# shorter view

# combining alien_no_sim and alien_re_sim

> NO_ALIE_002_sim <- rbind(data.frame(region = 'Norway', sim = alien.no_sim), alien_re_sim)

Reviewer: region should be called part.

Would be also good to add a column for year. Just repeat 2024 all the way down.

```

## 11. Export file

<!--# 

Optional: Display the code (don't execute it) or the workflow for exporting the indicator values to file. Ideally the indicator values are exported as a georeferenced shape or raster file with indicators values, reference values and errors. You can also chose to export the raw (un-normalised or unscaled variable) as a separate product. You should not save large spatial output data on GitHub. You can use eval=FALSE to avoid code from being executed (example below - delete if not relevant) 

-->

```{r export}
#| eval: false

# write to parquet using arrow 

> arrow::write_parquet(NO_ALIE_002_sim,"data/NO_ALIE_002_sim.parquet")

Reviewer: Nice :) This file is not committed yet I see.
```suggestion
arrow::write_parquet(NO_ALIE_002_sim,"data/resuls_NO_ALIE_002.parquet")
```
